### ISSUE TITLE
some question about the capacity of this codes ? 

### ISSUE THREAD

**Issue by zeynali** [pierluigiferrari/ssd_keras#79]
Hi,
Q1) is this possible to add or remove a few of layers ? How ?is this possible to freeze the layers ? how ?
Q2)**how do chose the name of layers** : for example , in the ssd512 models , 
` conv2_1 = Conv2D(128, (3, 3), activation='relu', padding='same', kernel_initializer='he_normal', kernel_regularizer=l2(l2_reg), name='conv2_1')(pool1)`

the name of layers must be same name in caffe ? if so , Where do you get the name of the layers?

Q3) my GPU is GTX 1080 (8G) and Ram 16G , Cpu i3 7100k , is this possible to retrain ssd512 on my own dataset?

Q4) what are they fc (6,7) ? is not ssd full convolution ? 

```
fc6 = Conv2D(1024, (3, 3), dilation_rate=(6, 6), activation='relu', padding='same', kernel_initializer='he_normal', kernel_regularizer=l2(l2_reg), name='fc6')(pool5)
 fc7 = Conv2D(1024, (1, 1), activation='relu', padding='same', kernel_initializer='he_normal', kernel_regularizer=l2(l2_reg), name='fc7')(fc6)


```

Q5) is that possible to retrain this models with TFrecord dataset file ? 

**Request)** please put the ssd512_training.ipynp.

**Comment by pierluigiferrari** [comment]
1. Adding, removing, and freezing layers works just as with any other Keras model. This is a general Keras question, it doesn't have anything to do with my SSD implementation in particular, so I would refer you to the [Keras API documentation](https://keras.io/) or Stackoverflow to learn how to add or freeze layers.
2. The layer names don't have to be the same as the Caffe layer names in general, but in my ported weights files I stuck with the original Caffe model layer names, so in order to load these weights `by_name`, the names of the layers in the model must be the same as the names of the respective layers in the HDF5 weights file. You get the names of a Caffe model from the `.prototxt` file that defines the Caffe model.
3. Yes.
4. SSD is fully convolutional. `fc6` and `fc7` are just called that to indicate that they were transformed into convolutional layers from the `fc6` and `fc7` layers of VGG.
5. This is also a general Keras question, it has nothing to do with my SSD implementation in particular. I've recently followed a few Keras and TensorFlow issues that discussed using `TFRecord` files with Keras models, but I don't know what the status is there. It is certainly possible by using this SSD implementation inside TensorFlow's Estimator API, which now fully supports Keras models.

**Comment by zeynali** [comment]
A1 : You can pass a `trainable` argument (boolean) to a layer constructor to set a layer to be non-trainable:
frozen_layer = Dense(32, trainable=False)

**Comment by pierluigiferrari** [comment]
Yes, you can freeze layers in any Keras model.

**Comment by zeynali** [comment]
Hi , 
i want to calculate mAP ,Precision and Recall metrics for inference .
and i want to view training/evaluation loss at the same time to realize when i stop the training and when happens Early Stopping .

**Comment by pierluigiferrari** [comment]
1. Q1 does not contain a question. Tools and tutorials for training and evaluation are all there.
2. Because they have two different sizes. You have to read the code and/or the paper.
